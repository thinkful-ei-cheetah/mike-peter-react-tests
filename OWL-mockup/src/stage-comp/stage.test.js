import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from './stage';
import STORE from '../store.js';

//this is the test case
it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  //render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Stage actors={STORE.participants} />, div);

  //clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Stage actors={STORE.participants} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});

it('renders the UI as expected with no unreads', () => {
  const tree = renderer
    .create(<Stage actors={STORE.participants} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
});