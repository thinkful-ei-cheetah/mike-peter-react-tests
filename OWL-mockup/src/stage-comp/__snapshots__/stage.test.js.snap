// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  className="stage"
>
  <div
    className="actor"
    id={1}
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <h3>
      Koren Templeton
    </h3>
    <p>
      actor
    </p>
  </div>
  
  
  
  <div
    className="actor"
    id={5}
  >
    <img
      alt="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <h3>
      Ashla Attwool
    </h3>
    <p>
      actor
    </p>
  </div>
  <div
    className="user"
    id="user"
  >
    <img
      alt="avatar"
    />
    <h3>
      test User
    </h3>
    <p>
      this is user
    </p>
  </div>
</section>
`;

exports[`renders the UI as expected with no unreads 1`] = `
<section
  className="stage"
>
  <div
    className="actor"
    id={1}
  >
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <h3>
      Koren Templeton
    </h3>
    <p>
      actor
    </p>
  </div>
  
  
  
  <div
    className="actor"
    id={5}
  >
    <img
      alt="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <h3>
      Ashla Attwool
    </h3>
    <p>
      actor
    </p>
  </div>
  <div
    className="user"
    id="user"
  >
    <img
      alt="avatar"
    />
    <h3>
      test User
    </h3>
    <p>
      this is user
    </p>
  </div>
</section>
`;
